src/saltext/vmware/states/esxi.py

```diff
@@ -145,6 +147,294 @@ def role_absent(
+                            host_name=host,
+                            service_instance=service_instance,
+                        )
+                        ret["changes"].update(ret_save)
+                    except salt.exceptions.SaltException as exc:
+                        hosts_in_error.append(host)
+                        sample_exception = str(exc)
+            ret["comment"] = "vmkernel adapter {} created on {}, updated on {}.".format(
+                name,
+                ",".join(set(add_on_hosts) - set(hosts_in_error)) or "-",
+                ",".join(set(update_on_hosts) - set(hosts_in_error)) or "-",
+            )
+            ret["result"] = True
+            if hosts_in_error:
+                ret["comment"] += "erred on {}. Sample exception - {}".format(
+                    ",".join(hosts_in_error), sample_exception
+                )
+                ret["result"] = False
+        return ret
+    except (
+        vim.fault.InvalidState,
+        vim.fault.NotFound,
+        vim.fault.HostConfigFault,
+        vmodl.fault.InvalidArgument,
+        salt.exceptions.VMwareApiError,
+        vim.fault.AlreadyExists,
+    ) as exc:
+        raise salt.exceptions.SaltException(str(exc))
+
+
+def vmkernel_adapter_absent(
+    name, datacenter_name=None, cluster_name=None, host_name=None, service_instance=None
+):
+    """
+    Ensure VMKernel Adapter exists on matching ESXi hosts.
+
+    name
+        The name of the VMKernel interface to update. (required).
+
+    datacenter_name
+        Filter by this datacenter name (required when cluster is specified)
+
+    cluster_name
+        Filter by this cluster name (optional)
+
+    host_name
+        Filter by this ESXi hostname (optional)
+
+    service_instance
+        Use this vCenter service connection instance instead of creating a new one. (optional).
+
+    .. code-block:: bash
+
+        salt '*' vmware_esxi.update_vmkernel_adapter dvswitch_name=dvs1 mtu=2000
+    """
+    log.debug("Running vmware_esxi.vmkernel_adapter_absent")
+    ret = {"name": name, "result": None, "comment": "", "changes": {}}
+    if not service_instance:
+        service_instance = get_service_instance(opts=__opts__, pillar=__pillar__)
+    hosts = utils_esxi.get_hosts(
+        service_instance=service_instance,
+        host_names=[host_name] if host_name else None,
+        cluster_name=cluster_name,
+        datacenter_name=datacenter_name,
+        get_all_hosts=host_name is None,
+    )
+
+    try:
+        adapters = __salt__["vmware_esxi.get_vmkernel_adapters"](
+            adapter_name=name,
+            datacenter_name=datacenter_name,
+            cluster_name=cluster_name,
+            host_name=host_name,
+            service_instance=service_instance,
+        )
+        delete_on_hosts = [h.name for h in hosts if adapters[h.name]]
+        if __opts__["test"]:
+            ret["comment"] = "vmkernel adapter {} will be deleted on {} host(s).".format(
```

Comment:
**comment** not sure if it matters but `{!r}` uses the string repr, meaning wrap in `''` (usually). It just depends on what the `name` might be if that's helpful at all.